includes/Plugin.php

```diff
@@ -87,6 +87,29 @@ function() {
 			return;
 		}
 
+		// In the case of IDNs, ensure the ASCII and non-ASCII domains
+		// are treated as allowable origins.
+		add_filter(
+			'allowed_redirect_hosts',
+			function( $hosts ) {
+				$wpp = wp_parse_url( home_url() );
+
+				// If this host isn't an IDN (or is already an ASCII-formatted
+				// IDN), we can return the existing `$hosts` array unmodified.
+				if ( idn_to_ascii( $wpp['host'], 0, INTL_IDNA_VARIANT_UTS46 ) === $wpp['host'] ) {
+					return $hosts;
+				}
+
+				// If this host is an IDN in Unicode format, we need to add the
+				// urlencoded versions of the domain to the `$hosts` array,
+				// because this is what will be used for redirects.
+				$hosts[] = rawurlencode( $wpp['host'] );
+
+				return $hosts;
+			},
+			1
```

Comment:
Running this early shouldn't be necessary as this returns a list of hosts – is this intended to reference the number of args the callback accepts? If so, this would be the 4th argument to `add_filter` and defaults to `1` in which case this should be omitted.

Reply:
Oops, that might've been from testing and probably isn't needed, cheers 👍🏻